WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -500,6 +495,16 @@ protected void onNewIntent(Intent intent) {
         super.onNewIntent(intent);
         setIntent(intent);
         AppLog.i(T.MAIN, "main activity > new intent");
+
+        if (FluxCUtils.isSignedInWPComOrHasWPOrgSite(mAccountStore, mSiteStore)) {
+            if (intent.hasExtra(ARG_NOTIFICATION_TYPE)) {
+                NotificationType notificationType =
+                        (NotificationType) intent.getSerializableExtra(ARG_NOTIFICATION_TYPE);
+                if (notificationType != null) {
+                    mSystemNotificationsTracker.trackTappedNotification(notificationType);
+                }
+            }
+        }
```

Comment:
this seems safe enough...TBH could not find the path that can trigger this code 🙃 ...happy to learn in case you want to share, thanks 🙇‍♂️

Reply:
Been testing this more extensively than I did when creating the issue / PR and found out to my surprise, that onCreate() in WPMainActivity is always called upon tapping on each new comment / comment like notification, thus the tracks information was good.
I was under the (wrong) impression that new intents from tapping on the notifications would bring the existing Activity to the foreground instead of creating a new instance every time, but turns out [`showWPComNotificationForBuilder()`](https://github.com/wordpress-mobile/WordPress-Android/blob/79ec027b58af46b6eb379e4fb6d5126c68720727/WordPress/src/main/java/org/wordpress/android/push/GCMMessageHandler.java#L732) will be passing these flags to the intent:
```
            resultIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK
                                  | Intent.FLAG_ACTIVITY_CLEAR_TASK);
```
Effectively this means creating a new WPMainActivity each time, and hence removing the code from `onCreate` and adding it to `onNewIntent` would incur in this new problem. Thank you for digging into this thoroughly @develric , and apologies for not testing this myself in the first place.

This change was reverted in cef1846da7 so, should be good for another pass now 👍